StarRocks can not insert to Hive4 tables

## Enhancement
Test env:
StarRocks latest 3.2.2 version, Hive master(or 4.0.0-beta-1).

I use SR3.2.2 to insert Hive4 tables, it will faill with exception:
`mysql> insert into hivecat.testdb.testsr values(1143);`
`ERROR 1064 (HY000): Failed to update following tasks: alter table parameters testsr
`

The detail from fe.log
`Caused by: org.apache.thrift.TApplicationException: Required field 'engine' is unset! Struct:TableStatsRequest(dbName:testdb, tblName:testsr, colNames:[id], engine:null)
        at org.apache.thrift.TServiceClient.receiveBase(TServiceClient.java:79) ~[libthrift-0.13.0.jar:0.13.0]
        at org.apache.hadoop.hive.metastore.api.ThriftHiveMetastore$Client.recv_get_table_statistics_req(ThriftHiveMetastore.java:4161) ~[hive-apache-3.1.2-13.jar:?]
        at org.apache.hadoop.hive.metastore.api.ThriftHiveMetastore$Client.get_table_statistics_req(ThriftHiveMetastore.java:4148) ~[hive-apache-3.1.2-13.jar:?]
        at org.apache.hadoop.hive.metastore.HiveMetaStoreClient.getTableColumnStatistics(HiveMetaStoreClient.java:735) ~[starrocks-fe.jar:?]
        at org.apache.hadoop.hive.metastore.HiveMetaStoreClient.getTableColumnStatistics(HiveMetaStoreClient.java:727) ~[starrocks-fe.jar:?]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_221]
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_221]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_221]
        at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_221]
        at org.apache.hadoop.hive.metastore.RetryingMetaStoreClient.invoke(RetryingMetaStoreClient.java:208) ~[hive-apache-3.1.2-13.jar:?]
        at com.sun.proxy.$Proxy32.getTableColumnStatistics(Unknown Source) ~[?:?]
        ... 22 more
`


The reason is that Hive4 added a `engine `field in hms thrift column stats api, which is not backward compatibility, and StarRocks use the TrinoDB HMS client which is compatible with Hive3 to access Hive4, so this exception will happen.

I have thought some workarounds to fix this issue:

1. First, StarRocks does't need to throw exception when invoking the hms column stats api, as stats is only used for CBO stage, and we can allow users to insert hive table and just log a error if getting the hms stats api unsuccessfully.

2. Second, i will file a ticket in Apache Hive, and try to fix the issue under the hood. You can to track this ticket https://issues.apache.org/jira/browse/HIVE-27984 .

--- a/fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveCommitter.java
+++ b/fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveCommitter.java
@@ -334,7 +334,7 @@ public class HiveCommitter {
             Joiner.on("; ").appendTo(message, failedUpdateStatsTaskDescs);
             StarRocksConnectorException exception = new StarRocksConnectorException(message.toString());
             suppressedExceptions.forEach(exception::addSuppressed);
-            throw exception;
+            LOG.error(exception);
         }
     }
 

--- a/fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveCommitter.java
+++ b/fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveCommitter.java
@@ -334,7 +334,13 @@ public class HiveCommitter {
             Joiner.on("; ").appendTo(message, failedUpdateStatsTaskDescs);
             StarRocksConnectorException exception = new StarRocksConnectorException(message.toString());
             suppressedExceptions.forEach(exception::addSuppressed);
-            LOG.error(exception);
+            // Insert into Hive4 table occur failure caused by compatibility issue between Hive3 and Hive4 thrift HMS client.
+            // Check https://github.com/StarRocks/starrocks/issues/38620 and HIVE-27984 for more details.
+            if (ConnectContext.get() != null && ConnectContext.get().getSessionVariable().enableHiveColumnStats()) {
+                throw exception;
+            } else {
+                LOG.error(exception);
+            }
         }
     }
 

